doc/adr/0004-global-snapshot-trigger.md

```diff
@@ -0,0 +1,23 @@
+# 4. Global snapshot trigger
+
+Date: 2022-01-05
+
+## Status
+
+Proposed
+
+## Context
+
+Global snapshot must have a deterministic trigger so then snapshots created on
+different nodes should converge (contains the same data) and pass the majority algorithm.
+
+As described in [ADR#2](./0002-global-and-state-channel-snapshots.md), Global
+snapshot contains both DAG blocks and StateChannel snapshots.
```

Comment:
In general this should be the case, but we can allow state channels to commit just blocks if they pay for the space in the L0 dag and L0 to validate their blocks